src/screens/DiscoveryScreenV2/fragment/DomainFragment.js

```diff
@@ -169,42 +181,71 @@ const DomainFragment = ({
     if (searchText.length > 0) fetchData();
   };
 
-  const __renderDiscoveryItem = (from, key, item, index) => {
+  const renderRecentSearch = (index) => {
     return (
-      <View key={`${key}-${index}`} style={styles.domainContainer}>
-        {route.name === 'Followings' && item.user_id_follower !== null && (
-          <DomainList
-            isDomain={true}
-            onPressBody={() => __handleOnPressDomain(item)}
-            handleSetFollow={() => __handleFollow(from, true, item, index)}
-            handleSetUnFollow={() => __handleFollow(from, false, item, index)}
-            item={{
-              name: item.domain_name,
-              image: item.logo,
-              isunfollowed: !item.following,
-              description: item.short_description || null
-            }}
-          />
-        )}
-        {route.name !== 'Followings' && (
-          <DomainList
-            isDomain={true}
-            onPressBody={() => __handleOnPressDomain(item)}
-            handleSetFollow={() => __handleFollow(from, true, item, index)}
-            handleSetUnFollow={() => __handleFollow(from, false, item, index)}
-            item={{
-              name: item.domain_name,
-              image: item.logo,
-              isunfollowed: !item.following,
-              description: item.short_description || null
-            }}
-          />
-        )}
-      </View>
+      index === 0 &&
+      !withoutRecent && (
+        <RecentSearch
+          shown={isFirstTimeOpen}
+          setSearchText={setSearchText}
+          setIsFirstTimeOpen={setIsFirstTimeOpen}
+        />
+      )
+    );
+  };
+
+  const renderItem = ({from, item, index}) => {
+    if (item.separator) {
+      return (
+        <>
+          {renderRecentSearch(index)}
+          <DiscoveryTitleSeparator text="Suggested Domains" key="domain-title-separator" />
+        </>
+      );
+    }
+
+    return (
+      <>
+        {renderRecentSearch(index)}
+        <View style={styles.domainContainer}>
+          {(route.name === 'Followings' && item.user_id_follower !== null) ||
+          route.name !== 'Followings' ? (
+            <DomainList
+              isDomain={true}
+              onPressBody={() => __handleOnPressDomain(item)}
+              handleSetFollow={() => __handleFollow(from, true, item, index)}
+              handleSetUnFollow={() => __handleFollow(from, false, item, index)}
+              item={{
+                name: item.domain_name,
+                image: item.logo,
+                isunfollowed: !item.following,
+                description: item.short_description || null
+              }}
+            />
+          ) : null}
+        </View>
+      </>
     );
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The refactoring of the rendering logic into separate functions (`renderRecentSearch`, `renderItem`) improves readability and maintainability of the code. It also allows for better separation of concerns.
```diff
- const __renderDiscoveryItem = (from, key, item, index) => {
-    return (
-     <View key={`${key}-${index}`} style={styles.domainContainer}>
-       {route.name === 'Followings' && item.user_id_follower !== null && (
-         <DomainList
-           isDomain={true}
-           onPressBody={() => __handleOnPressDomain(item)}
-           handleSetFollow={() => __handleFollow(from, true, item, index)}
-           handleSetUnFollow={() => __handleFollow(from, false, item, index)}
-           item={{
-             name: item.domain_name,
-             image: item.logo,
-             isunfollowed: !item.following,
-             description: item.short_description || null
-           }}
-         />
-       )}
-       {route.name !== 'Followings' && (
-         <DomainList
-           isDomain={true}
-           onPressBody={() => __handleOnPressDomain(item)}
-           handleSetFollow={() => __handleFollow(from, true, item, index)}
-           handleSetUnFollow={() => __handleFollow(from, false, item, index)}
-           item={{
-             name: item.domain_name,
-             image: item.logo,
-             isunfollowed: !item.following,
-             description: item.short_description || null
-           }}
-         />
-       )}
-     </View>
-    );
-  };
+ const renderRecentSearch = (index) => {
+   return (
+     index === 0 &&
+     !withoutRecent && (
+       <RecentSearch
+         shown={isFirstTimeOpen}
+         setSearchText={setSearchText}
+         setIsFirstTimeOpen={setIsFirstTimeOpen}
+       />
+     )
+   );
+ };
+ 
+ const renderItem = ({from, item, index}) => {
+   if (item.separator) {
+     return (
+       <>
+         {renderRecentSearch(index)}
+         <DiscoveryTitleSeparator text="Suggested Domains" key="domain-title-separator" />
+       </>
+     );
+   }
+ 
+   return (
+     <>
+       {renderRecentSearch(index)}
+       <View style={styles.domainContainer}>
+         {(route.name === 'Followings' && item.user_id_follower !== null) ||
+         route.name !== 'Followings' ? (
+           <DomainList
+             isDomain={true}
+             onPressBody={() => __handleOnPressDomain(item)}
+             handleSetFollow={() => __handleFollow(from, true, item, index)}
+             handleSetUnFollow={() => __handleFollow(from, false, item, index)}
+             item={{
+               name: item.domain_name,
+               image: item.logo,
+               isunfollowed: !item.following,
+               description: item.short_description || null
+             }}
+           />
+         ) : null}
+       </View>
+     </>
+    );
+  };
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->